File: app/classes/Deck.cs
public class Deck {
    private List<Card> cards;

    public Deck() {
        cards = new List<Card>();
    }

    public void AddCard(Card card) {
        cards.Add(card);
    }

    public Card DrawCard() {
        if (cards.Count == 0) {
            throw new InvalidOperationException("No cards left in the deck.");
        }
        Card drawnCard = cards[0];
        cards.RemoveAt(0);
        return drawnCard;
    }

    public void Shuffle() {
        if (cards.Count <= 1) return;
        Random rand = new Random();
        for (int i = 0; i < cards.Count; i++) {
            int j = rand.Next(i, cards.Count);
            (cards[i], cards[j]) = (cards[j], cards[i]);
        }
    }

    public int Count => cards.Count;

    public override string ToString() {
        return string.Join("\n", cards);
    }
}

File: app/tests/deck.py
import random

class Deck:
    def __init__(self):
        self.cards = []

    def add_card(self, card):
        self.cards.append(card)

    def draw_card(self):
        if not self.cards:
            raise ValueError("No cards left in the deck.")
        drawn_card = self.cards.pop(0)
        return drawn_card
    
    def shuffle(self):
        if len(self.cards) <= 1:
            return
        random.shuffle(self.cards)

    @property
    def count(self):
        return len(self.cards)
    
    def __str__(self):
        return "\n".join(str(card) for card in self.cards)